osu.Game.Rulesets.Osu/UI/DrawableOsuRuleset.cs

```diff
@@ -40,12 +41,17 @@ public DrawableOsuRuleset(Ruleset ruleset, IBeatmap beatmap, IReadOnlyList<Mod>?
         }
 
         [BackgroundDependencyLoader]
-        private void load(ReplayPlayer? replayPlayer)
+        private void load(Player? player, MultiSpectatorScreen? multiSpectatorScreen)
         {
-            if (replayPlayer != null)
+            if (player is ReplayPlayer || player is SpectatorPlayer)
             {
-                PlayfieldAdjustmentContainer.Add(new ReplayAnalysisOverlay(replayPlayer.Score.Replay));
-                replayPlayer.AddSettings(new ReplayAnalysisSettings(Config));
+                PlayfieldAdjustmentContainer.Add(new ReplayAnalysisOverlay(player.Score.Replay));
+
+                // if in multiplayer spectator mode, place the settings elsewhere
+                if (multiSpectatorScreen == null)
+                    player.AddSettings(new ReplayAnalysisSettings(Config));
+                else if (!multiSpectatorScreen.SettingsAdded)
+                    multiSpectatorScreen.AddSettings(() => new ReplayAnalysisSettings(Config));
```

Comment:
I was *sort of* fine with this when it only affected replay player but this I am not fine with. It's not sane.

If the replay analysis settings are to be displayed on the multiplayer spectator screen then they should be added there _by the multiplayer spectator screen itself_. Probably via some virtual method on `Ruleset` that normally does nothing.

Reply:
I agree with this, and in that case I think it'd make sense to have a non-ruleset-specific `AnalysisSettings`, which rulesets can create their own from and override a virtual method for creating the settings. If that is done, I think the `ReplayPlayer` should also add settings itself. Although, would it not make more sense to have the virtual method in `DrawableRuleset` as opposed to `Ruleset`?

Reply:
> Although, would it not make more sense to have the virtual method in `DrawableRuleset` as opposed to `Ruleset`?

We haven't done that sort of virtual in `DrawableRuleset` thus far. It would definitely need to be checked that constructing the drawable ruleset first to get the controls doesn't cause some massive overheads.

Reply:
Yeah that could cause a problem. On second thought anyway, I guess `DrawableRuleset` is more for gameplay drawing(?), while the analysis settings are separate from gameplay, so putting it in `Ruleset` seems fine.